flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBConnectorDeserializationSchema.java

```diff
@@ -110,12 +113,14 @@ public MongoDBConnectorDeserializationSchema(
             RowType physicalDataType,
             MetadataConverter[] metadataConverters,
             TypeInformation<RowData> resultTypeInfo,
-            ZoneId localTimeZone) {
+            ZoneId localTimeZone,
+            boolean preImageEnabled) {
         this.hasMetadata = checkNotNull(metadataConverters).length > 0;
         this.appendMetadataCollector = new AppendMetadataCollector(metadataConverters);
         this.physicalConverter = createConverter(physicalDataType);
         this.resultTypeInfo = resultTypeInfo;
         this.localTimeZone = localTimeZone;
+        this.hasPreImageData = preImageEnabled;
```

Comment:
I mean we can build different runtime converter  like `DeserializationRuntimeConverter`  (you can name a new one) for `hasPreImageData` enabled  case or  `hasPreImageData` disabled case during compile phase(i.e. constructor), and then we do not need to judge `hasPreImageData` per-record in deserialize method.

Reply:
Created `MongoDBConnectorFullChangelogDeserializationSchema` class and construct corresponding deserializer in `getScanRuntimeProvider`.